File: Lab 3/Java/SugestedProb8.java
import java.util.Scanner;

public class SugestedProb8 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt())
            return;
        int n = sc.nextInt();
        long[] a = new long[n];
        for (int i = 0; i < n; i++)
            a[i] = sc.nextLong();

        // Using int for sum as in Python example? "max_sum_var = -float('inf')".
        // Values can be large? I'll use long to be safe, but cast to int for print if
        // needed.
        // Python code: `sys.stdout.write(str(int(max_sum_var)))`

        long maxSum = Long.MIN_VALUE;
        long currentSum = 0;

        for (int i = 0; i < n; i++) {
            currentSum += a[i];
            if (currentSum > maxSum)
                maxSum = currentSum;
            if (currentSum < 0)
                currentSum = 0;
        }
        System.out.print(maxSum);
    }
}

File: Lab 3/SugestedProb8.py
import sys

data_var = sys.stdin.read().strip().split()

n_var = int(data_var[0])

a_var = []
i_var = 1
count_var = 0

while count_var < n_var:
    a_var.append(int(data_var[i_var]))
    count_var = count_var + 1
    i_var = i_var + 1

max_sum_var = -float('inf')
current_sum_var = 0

i_var = 0

while i_var < n_var:
    current_sum_var = current_sum_var + a_var[i_var]
    if current_sum_var > max_sum_var:
        max_sum_var = current_sum_var
    if current_sum_var < 0:
        current_sum_var = 0
    i_var = i_var + 1

sys.stdout.write(str(int(max_sum_var)))